### DISCUSSION TITLE
The name Result seems confusing to me

### DISCUSSION THREAD

**Discussion by BaruchMelio** [android/nowinandroid#830] [General]
I know it's minor but would like to hear your opinion about it.
The name "Result" implies a result of some specific operation and also conflicts with `kotlin.Result`.
The `Result` class in NIA seems to represent a continuous state rather than a result of a specific operation. In addition, It is used for the ui state so maybe there's some more appropriate name for it?

**Comment by dturner** [comment]
`Result` _is_ the result of some specific operation. It represents a single state which can either be `Success`, `Error` or `Loading`. What made you think it's a continuous state? 

As far as name collisions, this is an unfortunate side effect of naming things and is why we have namespaces.

> It is used for the ui state so maybe there's some more appropriate name for it? 

It's used to construct UI state inside ViewModels but is never exposed directly to the UI. Open to suggestions for a better name.

**Reply by BaruchMelio** [reply]
Well I don't think that `Loading` can be described as a result but rather some intermediate state of an operation that will eventually result in success or failure. One can say that an operation resulted in success or a failure but not that an operation resulted in loading. Therefore I would suggest a name like `LoadingResult` or `ResultState` just to distinguish it.
